### DISCUSSION TITLE
garden watering valves and soil sensors

### DISCUSSION THREAD

**Discussion by DarioKTS** [Koenkk/zigbee2mqtt#15016] [Devices]
Hey all, new memeber of the forum here :).

I've recently re-organised my home's lighting using zigbee2mqtt within a docker container, works flawlessly.
Now i'm onto the garden. i've noticed there aren't many soil sensors and watering valves in the device database.
I've been looking at these 2 :
[sensor](https://www.aliexpress.com/item/1005004762785020.html)
[valve](https://www.aliexpress.com/item/1005004021640604.html)

I'm more than willing to go through the process described in the ["support new devices" section](https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_1-pairing-the-device-with-zigbee2mqtt), and contribute to the device database, would just like to know if there is any chance that a device might outright not be compatible, so that i can switch to some other model before purchasing some unwanted paperweight..

thanks in advance

**Comment by JoppyFurr** [comment]
Hi,

At home we've got the same sensors, and made a start on getting things working. We don't have a Tuya hub, so are just getting the datapoints that it reports to a Zigbee receiver.

This is as far as we got, by having a best-guess at the meaning of the datapoints, which might help as a head start:

* Datapoint 3 - We guess this is the moisture, we've seen it drop to zero when removing the sensor from the soil
* Datapoint 5 - Appears to be temperature in °C
* Datapoint 14 - A status-byte? When it reports '1' the other values appear valid.
* Datapoint 15 - Maybe the battery percentage?

<pre>
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const tuya = require("zigbee-herdsman-converters/lib/tuya");
const utils = require("zigbee-herdsman-converters/lib/utils");
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;


const definition = 
    {
        fingerprint: tuya.fingerprint('TS0601', ['_TZE200_myd45weu']),
        model: 'QT-07S',
        vendor: 'TuYa',
        description: 'Soil moisture & temperature sensor',
        fromZigbee: [tuya.fzDataPoints],
        toZigbee: [tuya.tzDataPoints],
        configure: tuya.configureMagicPacket,
        exposes: [e.temperature(),e.soil_moisture(),e.battery()],
        meta: {
            tuyaDatapoints: [
                [3, 'soil_moisture', tuya.valueConverter.raw],
                [5, 'temperature', tuya.valueConverter.raw],
                [14, null, null], //status?
                [15, 'battery', tuya.valueConverter.raw], //battery?
            ],
        },
    }

module.exports = definition;
</pre>
